tests/integration/galley/analyze_test.go

```diff
@@ -180,6 +180,25 @@ func TestAllNamespaces(t *testing.T) {
 		})
 }
 
+func TestTimeout(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			g := NewGomegaWithT(t)
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-analyze",
+				Inject: true,
+			})
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			// We should time out immediately.
+			_, err := istioctlSafe(t, istioCtl, ns.Name(), true, "--timeout=0s")
```

Comment:
I would expect `0s` to disable the timeout?

Reply:
It doesn't, although perhaps it should. It's something I considered briefly but didn't implement. I wasn't sure that "set the timeout to 0 to disable it" would be any more intuitive to users than "set the timeout to an absurdly high number to (effectively) disable it."